components/org.wso2.carbon.identity.scim2.common/src/main/java/org/wso2/carbon/identity/scim2/common/impl/SCIMRoleManagerV2.java

```diff
@@ -0,0 +1,1138 @@
+                throw new BadRequestException("Role's permission change is not allowed for organizations.",
+                        ResponseCodeConstants.INVALID_VALUE);
+            }
+            updatePermissions(roleId, permissionOperations);
+        }
+        if (CollectionUtils.isNotEmpty(groupOperations)) {
+            updateGroups(roleId, groupOperations);
+        }
+        if (CollectionUtils.isNotEmpty(memberOperations)) {
+            updateUsers(roleId, currentRoleName, memberOperations);
+        }
+
+        HashMap<String, Boolean> requiredAttributes = new HashMap<>();
+        requiredAttributes.put(SCIMConstants.RoleSchemaConstants.DISPLAY_NAME_URI, true);
+        return getRole(roleId, requiredAttributes);
+    }
+
+    /**
+     * Filter users using multi-attribute filters or single attribute filters with pagination.
+     *
+     * @param node       Filter condition tree.
+     * @param startIndex Starting index of the count.
+     * @param count      Number of required results (count).
+     * @param sortBy     SortBy.
+     * @param sortOrder  Sort order.
+     * @return Detailed user list.
+     * @throws CharonException Error filtering the roles.
+     */
+    private RolesV2GetResponse filterRoles(Node node, Integer count, Integer startIndex, String sortBy,
+                                           String sortOrder)
+            throws CharonException, NotImplementedException, BadRequestException {
+
+        // Handle single attribute search.
+        if (node instanceof ExpressionNode) {
+            return filterRolesBySingleAttribute((ExpressionNode) node, count, startIndex, sortBy, sortOrder);
+        } else if (node instanceof OperationNode) {
+            String error = "Complex filters are not supported yet.";
+            throw new NotImplementedException(error);
+        } else {
+            throw new CharonException("Unknown operation. Not either an expression node or an operation node.");
+        }
+    }
+
+    /**
+     * Get the list of roles based on the filter.
+     *
+     * @param node       Expression node.
+     * @param startIndex Starting index.
+     * @param count      Number of results required.
+     * @param sortBy     SortBy.
+     * @param sortOrder  Sorting order.
+     * @return Filtered roles.
+     * @throws CharonException Error filtering the roles.
+     */
+    private RolesV2GetResponse filterRolesBySingleAttribute(ExpressionNode node, Integer count, Integer startIndex,
+                                                            String sortBy, String sortOrder)
+            throws CharonException, BadRequestException {
+
+        String attributeName = node.getAttributeValue();
+        String filterOperation = node.getOperation();
+        String attributeValue = node.getValue();
+        if (LOG.isDebugEnabled()) {
+            LOG.debug(
+                    "Filtering roles with filter: " + attributeName + " + " + filterOperation + " + " + attributeValue);
```

Comment:
Why do we have + as a string here?

Reply:
fixed in [9bb786c](https://github.com/wso2-extensions/identity-inbound-provisioning-scim2/pull/486/commits/9bb786c0be0dcc246e818ba8e2a962da88f0fb8b)